setcontext/main.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+
+import sys
+import pathlib
+
+#Add pytest
+#Build framework for  commandline tools.
+
+"""
+function to handle bad version string
+"""
+def validate_version_str(version_string: str):
+    #replace with regex
+    if not version_string.startswith("v"):
+        sys.exit(f"version string: {version_string} doesn't start with v")
+    if not version_string[1:].isdigit():
+        sys.exit(f"version string: {version_string} doesnt end in numerics 00 - 99")
+    if not len(version_string) == 4:
+        sys.exit(f"version string: {version_string} doesnt end in numerics 000 - 999")
```

Comment:
All of this can be condensed in one regex. Makes it easier to spot the expected format of version strings. 
```
import re
pat = re.compile("v\d\d\d$")
match = re.match(pat, version_string)
if not match:
   raise ValueError(<message>)
```

Reply:
Will also recommend using `raise ValueError` instead of `sys.exit`. If this module has the possibility of ever being imported in other modules, it is good practice to raise an Error and let the caller decide what to do with it.
For example, if some other script `foo.py` did this,
```
import main
valid = main.validate_version_string()
# we dont want the program to abruptly quit
# instead they could do

try:
   valid = main.validate_version_string()
except ValueError:
  # Take action
```